### DISCUSSION TITLE
Integrate Datascript (through Posh?)

### DISCUSSION THREAD

**Discussion by mtruyens** [day8/re-frame#697] [Ideas]
This would be awesome! 

See https://github.com/mpdairy/posh

**Comment by kristianmandrup** [comment]
@mtruyens  We are working on it ;) You are welcome to contribute!

https://gitter.im/metasoarous/posh

**Comment by mike-thompson-day8** [comment]
I'll be interested to see what people do.  But I'm going to close this issue for the moment.

**Comment by Conaws** [comment]
I've been using Re-Frame and Posh together quite a bit, but I think doing it much less effectively than I could.  So far I've been passing around a datascript connection to subscriptions and handlers the same way I would if it was any other value

```
(register-sub
 :e
 (fn [_ [_ conn eid]]
   (pull conn '[*] eid)))


(register-sub
 :text
 (fn [db]
   (reaction (:text @db))))

(register-sub
 :datoms
 (fn [_ [_ conn]]
   (q conn '[:find ?e ?attr ?val
             :where
             [?e ?attr ?val]])))

(register-sub
 :db-entities
 (fn [_ [_ conn]]
  (q conn '[:find ?e
             :where
             [?e]])))


(register-sub
 :tree
 (fn [db]
   (reaction (:tree @db))))
```

posh returns a reaction, so as you can see it's basically equivalent.

The thing I just saw over at

https://github.com/Day8/re-frame/issues/170

About the second argument most re-framers don't know about, seems to suggest I could be doing this much better.

Any suggestions on where I could look for inspiration?
